snowflake_driver.c

```diff
@@ -203,29 +202,20 @@ snowflake_handle_doer(pdo_dbh_t *dbh, const char *sql, size_t sql_len) /* {{{ */
     PDO_DBG_ENTER("snowflake_handle_doer");
     int ret = 0;
     pdo_snowflake_db_handle *H = (pdo_snowflake_db_handle *) dbh->driver_data;
-
-    // TODO add debugging statements
-
     PDO_DBG_INF("sql: %.*s, len: %d", sql_len, sql, sql_len);
     SF_STMT *sfstmt = snowflake_stmt(H->server);
     if (snowflake_query(sfstmt, sql, sql_len) == SF_STATUS_SUCCESS) {
-        PDO_DBG_INF("success");
         int64 rows = snowflake_affected_rows(sfstmt);
         if (rows == -1) {
-            pdo_stmt_t *pdostmt = ecalloc(1, sizeof(pdo_stmt_t));
-            pdostmt->driver_data = sfstmt;
-            pdostmt->dbh = dbh;
-            // TODO copy error from sfstmt to snowflake db handle
-            pdo_snowflake_error_stmt(pdostmt);
-            efree(pdostmt);
-            ret = -1; // TODO add ternary expression like: H->einfo.errcode ? -1 : 0
+            snowflake_propagate_error(H->server, sfstmt);
+            pdo_snowflake_error(dbh);
```

Comment:
Since snowflake_handler_doer doesn't have pdo_stmt_t, it has to replicate SF_ERROR from SF_STMT to SF_CONNECT.